async.md

```diff
@@ -220,4 +221,315 @@ Example
+			resp.error("caught: "+reason.message);
+		})
+~~~
+
+## ClearBladeAsync.FS.readFile(path,[encoding])
+
+ This function reads the entire contents of a file.
+ If encoding is not specified, contents will be a `UInt8Array`. If encoding is specified, contents will be a string.
+
+ * @param {string} path
+ * @param {string} encoding
+ * @returns {Promise} -  file contents
+
+Example
+
+~~~javascript
+	var FS = ClearBladeAsync.FS()
+	FS.readFile(path,[utf8]
+		.then(resp.success)
+		.catch(function(reason){
+			resp.error("caught: "+reason.message);
+		})
+~~~
+
+## ClearBladeAsync.FS.writeFile(path,data)
+
+This function writes the given data to a file, replacing the file if it already exists.
+
+ * @param {string} path
+ * @param {string|Uint8Array} data
+ * @returns {Promise}
+
+Example
+
+~~~javascript
+	var FS = ClearBladeAsync.FS()
+	FS.readFile(path,[utf8]
+		.then(resp.success)
+		.catch(function(reason){
+			resp.error("caught: "+reason.message);
+		})
+~~~
+
+## ClearBladeAsync.FS.renameFile(oldPath, newPath)
+
+ This function renames the file path.
+
+ * @param {string} oldPath
+ * @param {string} newPath
+ * @returns {Promise}
+
+Example
+
+~~~javascript
+	var FS = ClearBladeAsync.FS()
+	FS.renameFile(oldPath, newPath)
+		.then(resp.success)
+		.catch(function(reason){
+			resp.error("caught: "+reason.message);
+		})
+~~~
+
+## ClearBladeAsync.FS.copyFile(srcPath, dstPath)
+
+ This function copies srcPath to dstPath. It overwrites dstPath if it already exists.
+
+ * @param {string} srcPath
+ * @param {string} dstPath
+ * @returns {Promise}
+
+Example
+
+~~~javascript
+	var FS = ClearBladeAsync.FS()
+	FS.copyFile(srcPath, dstPath)
+		.then(resp.success)
+		.catch(function(reason){
+			resp.error("caught: "+reason.message);
+		})
+~~~
+
+## ClearBladeAsync.FS.deleteFile(path)
+
+This function deletes the file at the specified path.
+
+ * @param {string} path
+ * @returns {Promise}
+
+Example
+
+~~~javascript
+	var FS = ClearBladeAsync.FS()
+	FS.deleteFile(path)
+		.then(resp.success)
+		.catch(function(reason){
+			resp.error("caught: "+reason.message);
+		})
+~~~
+
+# File Stats 
+
+Class: ClearBladeAsync.FileStats()
+
+ This type definition for an object represents file metadata like size, permissions, etc.
+
+To instantiate the async file stats class just call:
+
+~~~javascript
+	var fstats = ClearBladeAsync.FileStats()
+~~~
+
+## ClearBladeAsync.FileStats.stat(path)
+
+The promise is resolved with the `FileStats` object for the given path.
+
+ * @param {string} path
+ * @returns {Promise}
+
+Example
+
+~~~javascript
+	var fstats = ClearBladeAsync.FileStats()
+	fstats.stat(path)
+		.then(resp.success)
+		.catch(function(reason){
+			resp.error("caught: "+reason.message);
+		})
+~~~
+
+# File Management 
+
+Class: ClearBladeAsync.File()
+
+This class adds support for interacting with the files in the sync’d filesystem.
+
+To instantiate the async file class just call:
+
+~~~javascript
+	var fstats = ClearBladeAsync.File()
```

Comment:
File needs both a deployment name and a file path on instantiation. should be ClearBladeAsync.File('myDeployment', 'sandbox/myFile.txt') or similar